src/configs/recommended.js

```diff
@@ -15,17 +15,8 @@ module.exports = {
     'primer-react/a11y-explicit-heading': 'error'
   },
   settings: {
-    github: {
-      components: {
-        Link: {props: {as: {undefined: 'a', a: 'a', button: 'button'}}},
-        Button: {default: 'button'}
-      }
-    },
     'jsx-a11y': {
-      components: {
-        Button: 'button',
-        IconButton: 'button'
-      }
+      components: require('./components')
```

Comment:
I created an individual file for component mapping (`components.js`). This allows us to consolidate all the mappings we want in one place. It also introduces a custom structure that is currently a proof of concept which should allow us to easily map composable components.

Would love additional 👀 on this!